File: tests/output/ruby/test/client/recommend_test.rb
require 'algolia'
require 'test/unit'

class TestClientRecommendClient < Test::Unit::TestCase
  include Algolia::Recommend
  # calls api with correct read host
  def test_api0
    client = Algolia::RecommendClient.create(
      'test-app-id',
      'test-api-key',
      { requester: Algolia::Transport::EchoRequester.new }
    )
    req = client.custom_get_with_http_info("test")
    assert_equal('test-app-id-dsn.algolia.net', req.host.url)
  end

  # calls api with correct write host
  def test_api1
    client = Algolia::RecommendClient.create(
      'test-app-id',
      'test-api-key',
      { requester: Algolia::Transport::EchoRequester.new }
    )
    req = client.custom_post_with_http_info("test")
    assert_equal('test-app-id.algolia.net', req.host.url)
  end

  # calls api with correct user agent
  def test_common_api0
    client = Algolia::RecommendClient.create(
      'APP_ID',
      'API_KEY',
      { requester: Algolia::Transport::EchoRequester.new }
    )
    req = client.custom_post_with_http_info("1/test")
    assert(req.headers['user-agent'].match(/^Algolia for Ruby \(\d+\.\d+\.\d+(-?.*)?\)(; [a-zA-Z. ]+ (\(\d+((\.\d+)?\.\d+)?(-?.*)?\))?)*(; Recommend (\(\d+\.\d+\.\d+(-?.*)?\)))(; [a-zA-Z. ]+ (\(\d+((\.\d+)?\.\d+)?(-?.*)?\))?)*$/))
  end

  # calls api with default read timeouts
  def test_common_api1
    client = Algolia::RecommendClient.create(
      'APP_ID',
      'API_KEY',
      { requester: Algolia::Transport::EchoRequester.new }
    )
    req = client.custom_get_with_http_info("1/test")
    assert_equal(2000, req.connect_timeout)
    assert_equal(5000, req.timeout)
  end

  # calls api with default write timeouts
  def test_common_api2
    client = Algolia::RecommendClient.create(
      'APP_ID',
      'API_KEY',
      { requester: Algolia::Transport::EchoRequester.new }
    )
    req = client.custom_post_with_http_info("1/test")
    assert_equal(2000, req.connect_timeout)
    assert_equal(30_000, req.timeout)
  end
end

File: tests/output/dart/test/client/recommend_test.dart
import 'package:algolia_client_recommend/algolia_client_recommend.dart';
import 'package:algolia_test/algolia_test.dart';
import 'package:test/test.dart';
import 'package:test_api/hooks.dart';

void main() {
  test('calls api with correct read host', () async {
    final requester = RequestInterceptor();
    final client = RecommendClient(
        appId: "test-app-id",
        apiKey: "test-api-key",
        options: ClientOptions(requester: requester));
    requester.setOnRequest((request) {
      expect(request.host.url, 'test-app-id-dsn.algolia.net');
    });
    try {
      final res = await client.customGet(
        path: "test",
      );
    } on InterceptionException catch (_) {
      // Ignore InterceptionException
    }
  });

  test('calls api with correct write host', () async {
    final requester = RequestInterceptor();
    final client = RecommendClient(
        appId: "test-app-id",
        apiKey: "test-api-key",
        options: ClientOptions(requester: requester));
    requester.setOnRequest((request) {
      expect(request.host.url, 'test-app-id.algolia.net');
    });
    try {
      final res = await client.customPost(
        path: "test",
      );
    } on InterceptionException catch (_) {
      // Ignore InterceptionException
    }
  });

  test('calls api with correct user agent', () async {
    final requester = RequestInterceptor();
    final client = RecommendClient(
      appId: 'appId',
      apiKey: 'apiKey',
      options: ClientOptions(requester: requester),
    );
    requester.setOnRequest((request) {
      TestHandle.current.markSkipped('User agent added using an interceptor');
    });
    try {
      final res = await client.customPost(
        path: "1/test",
      );
    } on InterceptionException catch (_) {
      // Ignore InterceptionException
    }
  });

  test('calls api with default read timeouts', () async {
    final requester = RequestInterceptor();
    final client = RecommendClient(
      appId: 'appId',
      apiKey: 'apiKey',
      options: ClientOptions(requester: requester),
    );
    requester.setOnRequest((request) {
      expect(5000, request.timeout.inMilliseconds);
    });
    try {
      final res = await client.customGet(
        path: "1/test",
      );
    } on InterceptionException catch (_) {
      // Ignore InterceptionException
    }
  });

  test('calls api with default write timeouts', () async {
    final requester = RequestInterceptor();
    final client = RecommendClient(
      appId: 'appId',
      apiKey: 'apiKey',
      options: ClientOptions(requester: requester),
    );
    requester.setOnRequest((request) {
      expect(30000, request.timeout.inMilliseconds);
    });
    try {
      final res = await client.customPost(
        path: "1/test",
      );
    } on InterceptionException catch (_) {
      // Ignore InterceptionException
    }
  });
}
